table/tables/index.go

```diff
@@ -463,3 +501,56 @@ func TryAppendCommonHandleRowcodecColInfos(colInfo []rowcodec.ColInfo, tblInfo *
 	}
 	return colInfo
 }
+
+type tempIndexKeyState byte
+
+const (
+	// KeyInTempIndexUnknown whether the key exists or not in temp index is unknown.
+	KeyInTempIndexUnknown tempIndexKeyState = iota
+	// KeyInTempIndexNotExist the key is not exist in temp index.
+	KeyInTempIndexNotExist
+	// KeyInTempIndexIsDeleted the key is marked deleted in temp index.
+	KeyInTempIndexIsDeleted
+	// KeyInTempIndexIsItself the key is correlated to itself in temp index.
+	KeyInTempIndexIsItself
+	// KeyInTempIndexConflict the key is conflict in temp index.
+	KeyInTempIndexConflict
+)
+
+// KeyExistInTempIndex is used to check the unique key exist status in temp index.
+func KeyExistInTempIndex(ctx context.Context, txn kv.Transaction, key kv.Key, distinct bool, h kv.Handle, IsCommonHandle bool) (tempIndexKeyState, kv.Handle, error) {
+	value, err := txn.Get(ctx, key)
+	if kv.IsErrNotFound(err) {
+		return KeyInTempIndexNotExist, nil, nil
+	}
+	if err != nil {
+		return KeyInTempIndexUnknown, nil, err
+	}
+
+	length := len(value)
+	// Firstly, we will remove the last byte of key version.
+	// It should be TempIndexKeyTypeBackfill or TempIndexKeyTypeMerge.
+	value = value[:length-1]
```

Comment:
Is it necessary to check the length? (to avoid panic because of the out-of-range error)